source/common/thread_local/thread_local_impl.h

```diff
@@ -32,9 +33,15 @@ class InstanceImpl : Logger::Loggable<Logger::Id::main>, public Instance {
     SlotImpl(InstanceImpl& parent, uint64_t index) : parent_(parent), index_(index) {}
     ~SlotImpl() override { parent_.removeSlot(*this); }
 
+    // non moveable.
```

Comment:
nit: I would just move these into NonCopyable, as IMO it's implied, or if you prefer make a new NonMoveable mix-in.

Reply:
Let's have a NonMoveable mixin and then a NonCopyableNonMoveable mixin?

Reply:
just learnt boost::noncopyable is also non-moveable. Make our noncopyable non-moveable as well.

Reply:
I think you can delete this now?